### PR TITLE
[#730] Refactor delete survey instance functionality

### PR DESCRIPTION
**Pull request by muloem**
* Adds a new queue for updating summary counts
* Delete a `SurveyedLocale` when there are no more related survey instances
* Remove test harness call

### CODE REVIEW

**Comment by iperdomo** [comment]
## Checklist

* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting

File: GAE/src/com/gallatinsystems/gis/map/MapUtils.java [hunk 1]
```diff
@@ -174,6 +175,11 @@ public static void recomputeCluster(Cache cache, SurveyedLocale locale, int delt
             }
         }
 
+        // delete locale if the Delta was a subtraction
+        SurveyedLocaleDao slDao = new SurveyedLocaleDao();
+        if (delta < 0) {
```
Comment (iperdomo) [inline GAE/src/com/gallatinsystems/gis/map/MapUtils.java]: Are we sure we want to delete a `SurveyLocale` on each substraction? I think we need to check for available `SurveyIntance` for this locale?

**Comment by iperdomo** [comment]
We need to update the copyright header for those modified files

File: GAE/src/com/gallatinsystems/gis/map/MapUtils.java (hunk 1 above)
Comment (muloem) [inline GAE/src/com/gallatinsystems/gis/map/MapUtils.java]: The deletion only happens when the deleted `SurveyInstance` is the only one connected to the `SurveyedLocale`

Comment (iperdomo) [inline GAE/src/com/gallatinsystems/gis/map/MapUtils.java]: This code is aware of that? The comment states:

     // delta can be +1 or -1, depending on if we want to add a point or delete a point.

So `delta` can be `-1`, but a `SurveyedLocale` can have more `SurveyInstance`s ?

Comment (muloem) [inline GAE/src/com/gallatinsystems/gis/map/MapUtils.java]: Yep. The task with the -1 delta is created here:
https://github.com/akvo/akvo-flow/pull/1086/files#diff-a521baebc25d5bbfb464614e3eb1f5daR684

See comment.

**Comment by iperdomo** [comment]
:+1:

**Pull request merged**